### PR TITLE
docs: Update the status of more feature to "under development"

### PR DESCRIPTION
**Pull request by prince-chrismc**
This is the last installment from me, the rest need a more expert review, I might make a meeting for this.

I think this can also close older PRs, they were stuck with both generators being experimental but with this PR one is deprecated and the other "under development trying not to break" which falls in the trap of https://github.com/conan-io/docs/pull/2269#issuecomment-949597141 but hopefully the more verbose language is enough 🤞 

closes #2268
closes #2269

### CODE REVIEW

**Comment by SSE4** [comment]
@prince-chrismc is it essential a rename `experimental` -> `under development` with slightly different wording; like, still a subject for breaking changes, but not very likely (however, still possible)?

**Comment by prince-chrismc** [comment]
Yes that's idea or the 2.0 features.

The stuff being removed will get marked deprecated :)

File: reference/config_files/settings.yml.rst [hunk 1]
```diff
@@ -241,13 +250,6 @@ This will be equivalent to the full version ``1913 (19.13)``. If even further de
 to the ``update`` subsetting in ``settings.yml``.
 
 
-clang
-+++++
-
-The release 13.0.0 will be released officially on September 21, 2021. However, Conan 1.40 will support it in settings.yml before
-the final release. It will be considered as **experimental** in case of incompatibility until the release.
```
Comment (RubenRBS) [inline reference/config_files/settings.yml.rst:248]: Nice catch!

**Review by RubenRBS — APPROVED** [review]

File: extending/hooks.rst [hunk 2]
```diff
@@ -3,9 +3,11 @@
 Hooks
 =====
 
+
 .. warning::
 
-    This is an **experimental** feature subject to breaking changes in future releases.
+    This is a **migrated** feature.  Please refer to the :ref:`Migration Guidelines<conan2_migration_guide>` learn about the changes.
```
Comment (memsharded) [inline extending/hooks.rst:9]: What does this "migrated" mean? The hooks in 1.X are not migrated from anywhere. Does it mean "this feature has already been migrated to 2.0 and it is available there"?

File: reference/conanfile/methods.rst [hunk 3]
```diff
@@ -1497,9 +1494,10 @@ current folder (the one containing the *conanfile.py*). The ``dst`` is relative
 generate()
 ----------
 
-.. warning::
+.. important::
```
Comment (memsharded) [inline reference/conanfile/methods.rst:1497]: ``generate`` as a method can be considered now stable enough, we won't break that.

**Review by memsharded — APPROVED** [review]
Good job

File: extending/hooks.rst (hunk 2 above)
Comment (prince-chrismc) [reply extending/hooks.rst:9]: It was not clear to me what the status was, i must have missed that in  my reset commit.

My understanding was an updated version exists in 2.0 already for uilianries PRs to docs.

I can remove it

File: reference/conanfile/methods.rst (hunk 3 above)
Comment (prince-chrismc) [reply reference/conanfile/methods.rst:1497]: 👍

File: extending/hooks.rst (hunk 2 above)
Comment (prince-chrismc) [reply extending/hooks.rst:9]:
```suggestion
    This is an **experimental** feature subject to breaking changes in future releases.
```

**Pull request merged**